src/ja/alphapolis/src/eu/kanade/tachiyomi/extension/ja/alphapolis/Alphapolis.kt

```diff
@@ -0,0 +1,262 @@
+            }
+        }
+        val hasNextPage = document.selectFirst("i.fa.fa-angle-double-right") != null
+        return MangasPage(mangas, hasNextPage)
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        val document = response.asJsoup()
+        return SManga.create().apply {
+            title = document.selectFirst(".manga-detail-description > .title > h1")!!.text()
+            val authors = document.select(".manga-detail-description .author-label .authors .mangaka").toList()
+            author = authors.filter { it.text().contains("原作") }
+                .mapNotNull { it.selectFirst("a")?.text() }
+                .joinToString()
+            artist = authors.filter { it.text().contains("漫画") }
+                .mapNotNull { it.selectFirst("a")?.text() }
+                .joinToString()
+            description = document.selectFirst(".manga-detail-outline .outline")?.text()
+            genre = document.select(".manga-detail-tags .official-manga-tags .official-manga-tag").joinToString { it.text() }
+            status = when (document.selectFirst(".wrap-content-status a[href*=complete]")?.text()) {
+                "連載中" -> SManga.ONGOING
+                "完結" -> SManga.COMPLETED
+                "休載中" -> SManga.ON_HIATUS
+                else -> SManga.UNKNOWN
+            }
+            thumbnail_url = document.selectFirst(".manga-bigbanner img")?.absUrl("src")
+        }
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        return document.select(".episode-list .episode-unit").map {
+            SChapter.create().apply {
+                setUrlWithoutDomain(it.selectFirst("a.read-episode")!!.absUrl("href"))
+                val titleText = it.selectFirst(".title")!!.text()
+                val isRental = it.selectFirst(".rental-coin") != null ||
+                    it.selectFirst(".icon-zero-yen") != null
+                if (isRental) name = "\uD83E\uDE99 $titleText" else name = titleText
+                date_upload = dateFormat.tryParse(it.selectFirst(".up-time")?.text())
+            }
+        }
+    }
+
+    override fun fetchPageList(chapter: SChapter): Observable<List<Page>> {
+        return Observable.fromCallable {
+            val parts = chapter.url.split("/")
+            val mangaId = parts[parts.size - 2].toInt()
+            val episodeId = parts.last().toInt()
+
+            val xsrfToken = getXsrfToken() ?: run {
+                client.newCall(GET(baseUrl + chapter.url, headers)).execute().close()
+                getXsrfToken()
+                    ?: throw Exception("XSRF-Token not found")
+            }
+
+            val viewerUrl = "$baseUrl/manga/official/viewer.json"
+
+            val newHeaders = super.headersBuilder()
+                .add("X-Requested-With", "XMLHttpRequest")
+                .add("X-XSRF-TOKEN", xsrfToken)
+                .set("Referer", baseUrl + chapter.url)
```

Comment:
You could possibly reuse logic:
```suggestion
                .set("Referer", getChapterUrl(chapter))
```